src/TurboMqtt/Client/ClientStreamOwner.cs

```diff
@@ -470,6 +470,9 @@ private async Task ReplaceTransport()
 
     protected override void PostStop()
     {
+        // force the transport to close - usually it will already be dead by now
+        _currentTransport?.AbortAsync();
```

Comment:
Force the transport to abort, even if we've already terminated it. Guarantees that we end up with the correct `ConnectionStatus` at the end.